packages/wallets/src/sdk.ts

```diff
@@ -0,0 +1,75 @@
+import type { EVMSmartWallet, EVMMPCWallet } from "@/evm";
+import type { SolanaSmartWallet, SolanaMPCWallet } from "@/solana";
+
+type EVMAdminSigner =
+    | {
+          type: "evm-keypair";
+          address: string;
+      }
+    | {
+          type: "evm-fireblocks-custodial";
+      }
+    | {
+          type: "evm-passkey";
+          id: string;
+          name: string;
+          publicKey: {
+              x: string;
+              y: string;
+          };
+      };
+
+type SolanaAdminSigner =
+    | {
+          type: "solana-keypair";
+          address: string;
+      }
+    | {
+          type: "solana-fireblocks-custodial";
+      };
+
+type WalletTypeToArgs = {
+    "evm-smart-wallet": [adminSigner: EVMAdminSigner, linkedUser: string];
+    "evm-mpc-wallet": [linkedUser: string];
+    "solana-smart-wallet": [adminSigner: SolanaAdminSigner, linkedUser: string];
+    "solana-mpc-wallet": [linkedUser: string];
+};
+
+class WalletSDK {
+    constructor(
+        private readonly apiKey: string,
+        private readonly jwt?: string
+    ) {}
+
+    public async getOrCreateWallet(
+        type: "evm-smart-wallet",
+        ...args: WalletTypeToArgs["evm-smart-wallet"]
+    ): Promise<EVMSmartWallet>;
+    public async getOrCreateWallet(
+        type: "evm-mpc-wallet",
+        ...args: WalletTypeToArgs["evm-mpc-wallet"]
+    ): Promise<EVMMPCWallet>;
+    public async getOrCreateWallet(
+        type: "solana-smart-wallet",
+        ...args: WalletTypeToArgs["solana-smart-wallet"]
+    ): Promise<SolanaSmartWallet>;
+    public async getOrCreateWallet(
+        type: "solana-mpc-wallet",
```

Comment:
Looks awkward, and a bit harder to maintain, but this will give the consumers great DX:

```ts
const wallet = await sdk.getOrCreateWallet("evm-smart-wallet", adminSigner: …);
//      ^ this will be typed as `EVMSmartWallet`               ^ this will have full autocomplete
```

Reply:
i think u can just make another type map like `WalletTypeToWallet` to use as the return type on the actual implementation func, and it should do the same thing without needing to define all these

Reply:
addressed below